src/core/ext/filters/client_idle/client_idle_filter.cc

```diff
@@ -27,12 +27,12 @@
 #include "src/core/lib/surface/channel_init.h"
 #include "src/core/lib/transport/http2_errors.h"
 
-// The idle filter is enabled in client channel by default.
-// Set GRPC_ARG_CLIENT_IDLE_TIMEOUT_MS to [1000, INT_MAX) in channel args to
-// configure the idle timeout.
-#define DEFAULT_IDLE_TIMEOUT_MS (30 /*minutes*/ * 60 * 1000)
-// The user input idle timeout smaller than this would be capped to it.
-#define MIN_IDLE_TIMEOUT_MS (1 /*second*/ * 1000)
+// The idle filter is disabled in client channel by default.
```

Comment:
This change will actually revert us to where this filter is disabled by default, which could potentially break things if people are depending on the idleness behavior.

Note that we can't address this by keeping the filter enabled if we revert the change from mutex to atomics, because QC's experiments showed that the mutex implementation affected performance.

Can we instead find a way to fix the problem without reverting?  I'm not an atomics expert, but maybe @soheilhy can suggest a fix.

Reply:
In my opinion, the performance difference is not so significant. If there is no easy fix in a week, I would prefer rolling back to mutex.

Reply:
@nanahpang Can you provide some quantifiable results on whether or not the mutex version is a significant performance hit? That was the other thing that surprised me about #19727 - I didn't see performance numbers posted in the PR (as folks like @soheilhy and @arjunroy typically do in atomic and other optimization PRs)

Reply:
In fact, let me go one step further. https://github.com/grpc/grpc/pull/19727/checks?check_run_id=195764367 shows that there were no improvements in microbenchmarks caused by this change and there were _several_ microbenchmarks with degradations. It seems like #19727 was an interesting design experiment but really should never have been merged.

Reply:
#19727 enabled the filter by default; it was not enabled before that, so performance comparisons before and after that PR are meaningless.

There was significant internal benchmarking and analysis of this feature before it was enabled.

Reply:
@vjpai Right, it doesn't show improvement in check. But since it is compared performance with and without the idleness filter, it is reasonable that we only see the overhead of implementing idleness filter as a result. 
I remember QC has written a microbenchmark to compare the performance of different filter implementations. @AspirinSJL Did he merge that, or it is just for experiments?

Reply:
For the existing benchmarks, using mutex instead of atomic locks has an (expected) reduction in use of CAS and an (expected) increase in the use of mutexes, but no significant CPU time difference for any ping-pong workload (https://github.com/grpc/grpc/pull/20854/checks?check_run_id=282184419).